protocol/contracts/oracle/src/state/supported_pairs.rs

```diff
@@ -178,6 +189,26 @@ where
     }
 }
 
+struct DefinitionVisitor;
+
+impl AnyVisitor for DefinitionVisitor {
+    type Output = CurrencyDefinition;
+
+    type Error = platform::never::Never;
+
+    fn on<C>(self) -> AnyVisitorResult<Self>
+    where
+        C: Currency + Serialize + DeserializeOwned,
```

Comment:
Are you sure we need `Serialize + Deserialize`

Reply:
It came with the signature as far as I am aware. Most likely the IDE generated the method for me, thus it's already a bound on the trait. I can remove it here, but it won't make a big difference overall.

Reply:
how big the difference is we cannot measure but for sure we have to limit the required bounds to their minimum

Reply:
I'll remove it from here, but it's the trait that needs to be reworked is what I meant. ;)